### DISCUSSION TITLE
Thick outline around image with

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#4810] [Digital Image Processing]
Centos 8, IM v7

I want to get a thick white or color outline around a image like this.

Is there a way?

Thanks,

![images](https://user-images.githubusercontent.com/12046302/152599239-4e80b6c9-f884-406f-a69c-e551c7a8fa0b.jpeg)

**Comment by fmw42** [comment]
See my script, contour, at http://www.fmwconcepts.com/imagemagick

**Reply by usmanpakistan** [reply]
[](url)Hi @fmw42 
Awesome script. btw. Need your help with an issue
**Project Background:** I am trying to make die cut stickers. The outline needs to be smooth with no jagged edges. This will allow cutter to easily cut the sticker.
**Issue**
I am trying to use your script to generate cutline on PNG Images(Images have transparent background). However, after a lot of tweaking I am still unable to manage 2 things

1. Empty area inside images (multiple components). I need the outline to be around the whole image. even with the image above, you can see empty space between the hear and the panda
2. Smooth edges. In images with finer detail the script should generate smooth outline. Please see the 3 attached image. 2nd image is the output from contour script. 3rd image is the required output

![imput Image](https://github.com/user-attachments/assets/0205aa31-b04f-4809-90de-78b231d84f3d)
![output from script](https://github.com/user-attachments/assets/c89fdf6c-3cc4-434b-9d92-b5917b50a262)

**Reply by usmanpakistan** [reply]
![required](https://github.com/user-attachments/assets/70bfae05-5d87-42ad-9294-9fc9fd747da6)
This is the output needed

**Reply by fmw42** [reply]
Please provide your exact command line that you used to process the image with my script.

**Reply by usmanpakistan** [reply]
sh contour.sh -d 10 test.png test3.png

**Comment by srini8080** [comment]
Wow! excellent
Thanks,
    On Saturday, February 5, 2022, 02:21:12 AM GMT+5:30, Fred Weinhaus ***@***.***> wrote:  
 
 


See my script, contour, at http://www.fmwconcepts.com/imagemagick

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by fmw42** [comment]
Is this what you want?

`contour -a 0 -d 30 -h off -r 10 -s 80 -o "gray(225)" image.png x.png`

![x](https://github.com/user-attachments/assets/317c0643-841e-4b8e-a4bb-c65432a0fd3f)

**Reply by usmanpakistan** [reply]
This is much better. However one limitation of sticker cutters is that they cannot move on sharp corners. The corners needs to smooth out
![402290549-317c0643-841e-4b8e-a4bb-c65432a0fd3f](https://github.com/user-attachments/assets/a071349d-374f-495a-991f-0aee4788155c)

**Reply by fmw42** [reply]
Increase -d argument (dilate) and then add -e argument (erode).  

`contour -a 0 -d 50 -e 20 -h off -r 10 -s 80 -o "gray(225)" image.png x.png`

![x](https://github.com/user-attachments/assets/43efef2a-5ea6-412d-b86b-a4abcb35c030)

**Reply by srini8080** [reply]
Wow! perfect, this is what I am looking for,

**Reply by fmw42** [reply]
Please post your input image and describe what is wrong with the result from the command that you show. Also identify your IM version and platform/OS.  Note that this script does not run on native Windows. You need to use Linux, Mac OSX, or Windows Unix or Cygwin.

**Reply by fmw42** [reply]
Add .sh to the end of the script name. (download from my web site)

contour.sh -a 0 -d 50 -e 20 -h off -r 10 -s 80 -o "gray(225)" image.png x.png

be sure you have Imagemagick in your $PATH environment variable so it can be found by the script.

See my home page Pointers for Use for configuration

[contour.zip](https://github.com/user-attachments/files/20051485/contour.zip)

**Reply by fmw42** [reply]
Yes add .sh as I just said to you above

**Reply by fmw42** [reply]
The arguments are dependent on the size of the image dimensions.

**Reply by fmw42** [reply]
Did it work?
